### ISSUE TITLE
Map.ActorsInWorld appears to include Player in its Actor[] return array

### ISSUE THREAD

**Issue by lovalmidas** [OpenRA/OpenRA#13006]
Map.ActorsInWorld appears to include at least one Player entity, which can lead to unexpected behaviour when attempting to perform certain functions on this array.

-------------------------------------

**To reproduce:**
Run this Lua function in a map with a "Germany" player

```
function DeleteBaseNodeActors () -- any name will do
  Trigger.AfterDelay(0, function()
    Utils.Do(Map.ActorsInWorld, function(a) 
      if a.IsDead then return end
      if a.Owner.Name == "Germany" then 
        if a.HasProperty("Destroy") then 
          a.Stop()
          a.Kill() 
        end
      end
    end)
  end)
end
```

This will return an error:
... ---> Eluant.LuaException: [string "BindingSupport.lua"]:30: Actor 'player' does not define a property 'Kill'

-------------------------------------

Using 

`Map.ActorsInBox(Map.TopLeft, Map.BottomRight, function(a) return true end)`

in place of

`Map.ActorsInWorld`

resolves this issue.

-------------------------------------
The problem can be made more serious by using a.Destroy() instead of a.Kill(). Attempting to call a Player (any player) value after using Destroy() on a Player results in a crash:

`Exception of type `System.NullReferenceException`: Object reference not set to an instance of an object.
   at OpenRA.Mods.Cnc.Effects.GpsDot.OpenRA.Effects.IEffectAboveShroud.RenderAboveShroud(WorldRenderer wr)
   at OpenRA.Graphics.WorldRenderer.<Draw>m__9(IEffectAboveShroud e)
   at System.Linq.Enumerable.<SelectManyIterator>d__16`2.MoveNext()
   at System.Linq.Enumerable.<ConcatIterator>d__58`1.MoveNext()
   at System.Linq.Enumerable.<ConcatIterator>d__58`1.MoveNext()
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at OpenRA.Graphics.WorldRenderer.Draw()
   at OpenRA.Game.RenderTick()
   at OpenRA.Game.Loop()
   at OpenRA.Game.Run()
   at OpenRA.Program.Run(String[] args)
   at OpenRA.Program.Main(String[] args)`

**Comment by Wobling** [comment]
So ActorsInWorld should be filtered for players before returning or players should never be in this array?

**Comment by lovalmidas** [comment]
It is strange to see a Player element in an array documented to be Actor[]. There are several Actor commands (e.g. Kill(), Destroy()) that tend to do bad things if done to a Player, and you can't call player.Kill() anyway except through this loophole. 

If a player (player 1) is subject to Destroy() through the method above, making a reference to a different player (e.g. player 2) will throw both a Lua error _and_ a crash (NullReferenceException as described above). The lua error will indicated that you tried to get a trait from 'player 1 (not in world)', even if the player you are referencing is player 2. 

If there is a reason for including Player it would be good to hear it.

**Comment by Wobling** [comment]
I wasn't sure if players were considered Actors, I couldn't see them inheriting but they do have an Actor Field.

public readonly Actor PlayerActor;

Would be good to get some overview from someone who's more familiar with this, I don't mind making the fix if I know what's expected :)

**Comment by GraionDilach** [comment]
Everything with traits are an actor, not just the units alone (or units which don't even enter the field, like the spy-boon proxy actors), but players and even the world itself has an actor definition.

Ofcourse, these actors probably shouldn't populate the ActorsInWorld Lua query.

**Comment by WolfGaming** [comment]
I would second @GraionDilach point here as players are not in the world in layman terms. (Even if they technically are you get my point right?)

**Comment by lovalmidas** [comment]
Since @GraionDilach hinted on an actor definition on the world, it might be worthwhile to check what kinds of actors is populated by ActorsInWorld. Probably the world itself is not included, but my guess is that solving PlayerActor may just divert the issue to some other WeirdActor.

Then we can consider what solution to take (especially if we discover more things in there). This could be resolved in the Lua script through checking of case scenarios, so this can be delegated to the responsibility of the Lua scripter. But at the least this should be documented.

For most scenarios, Using Map.ActorsInBox(_insert code for full map_) in place of Map.ActorsInWorld will resolve this issue.

**Comment by GraionDilach** [comment]
ActorsInBox is bad, because it'll only work for rectangular maps,the solution fails on isometric terrain (it also takes heightlevels in accordance).
